allennlp/tests/semparse/worlds/wikitables_variable_free_world_test.py

```diff
@@ -82,128 +94,113 @@ def test_get_valid_actions_returns_correct_set(self):
                                 ['count'])
 
         # These are the columns in table, and are instance specific.
-        check_productions_match(valid_actions['l'],
-                                ['fb:row.row.year',
-                                 'fb:row.row.league',
-                                 'fb:row.row.avg_attendance',
-                                 'fb:row.row.division',
-                                 'fb:row.row.regular_season',
-                                 'fb:row.row.playoffs',
-                                 'fb:row.row.open_cup'])
+        check_productions_match(valid_actions['y'],
+                                ['date_column:year'])
+
+        check_productions_match(valid_actions['f'],
+                                ['number_column:avg_attendance',
+                                 'number_column:division'])
+
+        check_productions_match(valid_actions['t'],
+                                ['string_column:league',
+                                 'string_column:playoffs',
+                                 'string_column:open_cup',
+                                 'string_column:regular_season'])
 
```

Comment:
What about `g` and `c`?  Where are their productions?  You're going to need them, because you have things like `r -> [<r,<g,r>>, r, g]` (so `g` shows up as a nonterminal).

Reply:
As I told you about this earlier today, they do not have productions of their own. There is still the issue that they show up as nonterminals while searching in `ActionSpaceWalker`, and converting action sequences to logical forms in `World`. I dealt with `MultiMatchNamedBasicTypes` separately in those classes (in the upcoming PR).

Reply:
It's also an issue in the model itself.  If they show up in the `ActionSpaceWalker`, they will show up in the model.  This doesn't work with how our models are structured, and it needs to change.

Reply:
How about we discuss it when we see the issue with the model? I wouldn't mind changing the way I implemented the functionality here if I have to then. The simpler alternative of adding the same name to the local mapping with multiple types would have been messy for the reason I stated above.

Reply:
Fine, we can change it later, but something's going to have to change, as we need to be able to consider all column types in a single timestep.